src/lib/alloc.c

```diff
@@ -429,12 +438,40 @@ static void *rmalloc_runtime(int zone, uint32_t caps, size_t bytes)
 		ptr = cache_to_uncache(ptr);
 
 	return ptr;
+}
+
+static void *rmalloc_runtime(int zone, uint32_t caps, size_t bytes)
+{
+	struct mm_heap *heap;
+	struct mm_heap *current = NULL;
+	void *ptr = NULL;
+
+	/* check runtime heap for capabilities */
+	do {
+		heap = get_runtime_heap_from_caps(caps, current);
```

Comment:
If first call get_runtime_heap_from_caps() returns a heap but rmalloc_runtime_get() fails, the second call of get_runtime_heap_from_caps() will always get the same heap from the first one?

Reply:
yes, that's the intention, to get the same heap and dump it status.